src/sql/src/ast/mod.rs

```diff
@@ -9,5 +9,39 @@
 
 //! SQL abstract syntax tree.
 
+use mz_sql_parser::ast::visit::Visit;
 pub use mz_sql_parser::ast::*;
 pub mod transform;
+
+/// A visitor that determines if object references are present.
```

Comment:
I think this is slightly incomplete because `SHOW` commands aren't "object references".